File: CPP/Program/PetShop.cpp
#include <string>

using namespace std;

class PetShop {
protected:
    int id;
    string nama_produk;
    double harga_produk;
    int stok_produk;

public:
    // constructors
    PetShop() : id(0), nama_produk(""), harga_produk(0.0), stok_produk(0) {} // default constructor
    PetShop(int id, string nama_produk, double harga_produk, int stok_produk)
        : id(id), nama_produk(nama_produk), harga_produk(harga_produk), stok_produk(stok_produk) {}

    // getters
    int getId() { return id; }
    string getNamaProduk() { return nama_produk; }
    double getHargaProduk() { return harga_produk; }
    int getStokProduk() { return stok_produk; }

    // setters
    void setId(int id) { this->id = id; }
    void setNamaProduk(string nama_produk) { this->nama_produk = nama_produk; }
    void setHargaProduk(double harga_produk) { this->harga_produk = harga_produk; }
    void setStokProduk(int stok_produk) { this->stok_produk = stok_produk; }
};

File: Java/Program/PetShop.java

class PetShop {
    protected int id;
    protected String namaProduk;
    protected double hargaProduk;
    protected int stokProduk;

    // Constructors
    public PetShop() {
        this.id = 0;
        this.namaProduk = "";
        this.hargaProduk = 0.0;
        this.stokProduk = 0;
    }

    public PetShop(int id, String namaProduk, double hargaProduk, int stokProduk) {
        this.id = id;
        this.namaProduk = namaProduk;
        this.hargaProduk = hargaProduk;
        this.stokProduk = stokProduk;
    }

    // getters
    public int getId() {
        return id;
    }

    public String getNamaProduk() {
        return namaProduk;
    }

    public double getHargaProduk() {
        return hargaProduk;
    }

    public int getStokProduk() {
        return stokProduk;
    }

    // setters
    public void setId(int id) {
        this.id = id;
    }

    public void setNamaProduk(String namaProduk) {
        this.namaProduk = namaProduk;
    }

    public void setHargaProduk(double hargaProduk) {
        this.hargaProduk = hargaProduk;
    }

    public void setStokProduk(int stokProduk) {
        this.stokProduk = stokProduk;
    }
}
